test/gracefulstop_test.go

```diff
@@ -42,7 +42,9 @@ type delayListener struct {
 	closeCalled  chan struct{}
 	acceptCalled chan struct{}
 	allowCloseCh chan struct{}
+	mu           sync.Mutex
```

Comment:
it looks like we are introducing mutex only to update `closed`? Then it should be put right above it indicating that anything below is guarded by mutex.

Reply:
Done